packages-exp/auth-exp/src/core/persistence/react_native.ts

```diff
@@ -25,36 +25,50 @@ import {
 } from './';
 
 /**
- * Persistence class that wraps AsyncStorage imported from `react-native` or `@react-native-community/async-storage`.
+ * Returns a persistence class that wraps AsyncStorage imported from
+ * `react-native` or `@react-native-community/async-storage`.
+ *
+ * Creates a "new"-able subclass on the fly that has an empty constructor.
+ *
+ * In the _getInstance() implementation (see src/core/persistence/index.ts),
+ * we expect each "externs.Persistence" object passed to us by the user to
+ * be able to be instantiated (as a class) using "new". That function also
+ * expects the constructor to be empty. Since ReactNativeStorage requires the
+ * underlying storage layer, we need to be able to create subclasses
+ * (closures, esentially) that have the storage layer but empty constructor.
  */
-export class ReactNativePersistence implements Persistence {
-  readonly type: PersistenceType = PersistenceType.LOCAL;
 
-  constructor(private readonly storage: ReactNativeAsyncStorage) {}
+export function makeReactNativePersistence(
```

Comment:
nit: we should call this reactNativePersistenceGenerator or Provider to match the other places we have providers, alternatively call "getReactNativePersistence" to match getMultiFactorResolver naming

Reply:
Done